src/Language/PureScript/Publish.hs

```diff
@@ -300,111 +297,93 @@ getResolvedDependencies declaredDeps = do
---   `bower list --json --offline`
-asResolvedDependencies :: Parse BowerError [(PackageName, DependencyStatus)]
+-- | Extracts all dependencies and their versions from a "resolutions" file, which
+-- is based on the output of `bower list --json --offline`
+asResolvedDependencies :: Parse D.ManifestError [(PackageName, (Maybe FilePath, DependencyStatus))]
 asResolvedDependencies = nubBy ((==) `on` fst) <$> go
   where
   go =
     fmap (fromMaybe []) $
       keyMay "dependencies" $
-        (++) <$> eachInObjectWithKey parsePackageName asDependencyStatus
+        (++) <$> eachInObjectWithKey parsePackageName asDirectoryAndDependencyStatus
              <*> (concatMap snd <$> eachInObject asResolvedDependencies)
 
--- | Extracts only the top level dependency names from the output of
---   `bower list --json --offline`
-asToplevelDependencies :: Parse BowerError [PackageName]
+-- | Extracts only the top level dependency names from a resolutions file.
+asToplevelDependencies :: Parse D.ManifestError [PackageName]
 asToplevelDependencies =
   fmap (map fst) $
     key "dependencies" $
       eachInObjectWithKey parsePackageName (return ())
 
-asDependencyStatus :: Parse e DependencyStatus
-asDependencyStatus = do
+asDirectoryAndDependencyStatus :: Parse e (Maybe FilePath, DependencyStatus)
+asDirectoryAndDependencyStatus = do
   isMissing <- keyOrDefault "missing" False asBool
   if isMissing
     then
-      return Missing
-    else
-      key "pkgMeta" $
+      return (Nothing, Missing)
+    else do
+      directory <- key "canonicalDir" asString
+      status <- key "pkgMeta" $
         keyOrDefault "_resolution" NoResolution $ do
           type_ <- key "type" asText
           case type_ of
             "version" -> ResolvedVersion <$> key "tag" asText
             other -> return (ResolvedOther other)
+      return (Just directory, status)
 
 warnUndeclared :: [PackageName] -> [PackageName] -> PrepareM ()
 warnUndeclared declared actual =
   traverse_ (warn . UndeclaredDependency) (actual \\ declared)
 
-handleDeps ::
-  [(PackageName, DependencyStatus)] -> PrepareM [(PackageName, Version)]
+handleDeps
+  :: [(PackageName, (Maybe FilePath, DependencyStatus))]
+  -> PrepareM [(PackageName, (FilePath, Version))]
 handleDeps deps = do
-  let (missing, noVersion, installed) = partitionDeps deps
+  let (missing, noVersion, installed, missingPath) = partitionDeps deps
   case missing of
     (x:xs) ->
       userError (MissingDependencies (x :| xs))
     [] -> do
       traverse_ (warn . NoResolvedVersion) noVersion
-      withVersions <- catMaybes <$> traverse tryExtractVersion' installed
-      filterM (liftIO . isPureScript . bowerDir . fst) withVersions
+      traverse_ (warn . MissingPath) missingPath
+      catMaybes <$> traverse tryExtractVersion' installed
 
   where
-  partitionDeps = foldr go ([], [], [])
-  go (pkgName, d) (ms, os, is) =
+  partitionDeps = foldr go ([], [], [], [])
+  go (pkgName, (Nothing, _)) (ms, os, is, mp) =
```

Comment:
I really don't like this code (which I take full responsibility for), btw. It's way too easy to miss things out accidentally. If you can think of a better way of doing it I'm all ears.